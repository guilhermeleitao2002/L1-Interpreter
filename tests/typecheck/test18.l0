type Point2D = struct { #x: int, #y: int };
type Point3D = struct { #x: int, #y: int, #z: int };

let process2DList:list<Point2D>->int = fn lst:list<Point2D> => {
  match lst {
    nil -> 0
    | h::t -> h.#x + h.#y
  }
};

let points3D:list<Point3D> = { #x=1, #y=2, #z=3 } :: nil;
points3D;;