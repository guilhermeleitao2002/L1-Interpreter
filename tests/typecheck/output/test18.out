TypeError: List element type mismatch: expected int, got struct { x: int, y: int, z: int }